src/Build.OM.UnitTests/Construction/ProjectTargetElement_Tests.cs

```diff
@@ -339,6 +343,35 @@ public void SetReturns()
             Assert.True(project.HasUnsavedChanges);
         }
 
+        /// <summary>
+        /// Parse invalid property under target
+        /// </summary>
+        [Fact]
+        public void ReadInvalidPropertyUnderTarget()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                ChangeWaves.ResetStateForTests();
+                env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", ChangeWaves.Wave17_6.ToString());
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                var error= Assert.Throws<InvalidProjectFileException>(() =>
+                {
+                    string projectFile = @"
+                    <Project>
+                        <Target Name='t'>
+                            <test>m</test>
+                        </Target>
+                    </Project>";
+
+                    TransientTestFile file = env.CreateFile("proj.csproj", projectFile);
+                    ProjectCollection collection = new ProjectCollection();
+                    collection.LoadProject(file.Path).Build().ShouldBeTrue();
+                });
+                error.ErrorCode.ShouldBeLessThanOrEqualTo("MSB4070");
```

Comment:
Shouldn't this be an exact match?

Reply:
Yeah, I misunderstand this equal function. Have updated to 
if (ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_6))
                {
                    error.ErrorCode.ShouldMatch("MSB4070");
                }
                else
                {
                    error.ErrorCode.ShouldMatch("MSB4067");
                }